app/models/level.rb

```diff
@@ -8,4 +8,28 @@ class Level < ApplicationRecord
 
   validates_uniqueness_of :mark, scope: :rubric_criterion_id
   validates_numericality_of :mark, greater_than_or_equal_to: 0
+
+  before_destroy :destroy_associated_marks
+  before_update :update_associated_marks
+
+  def destroy_associated_marks
+    marks = self.rubric_criterion.marks
```

Comment:
@ashih2018 change this to the "one line" version you were referring to in our meeting yesterday. I want to see exactly what happens.